OpenXmlPowerTools/WmlComparer/ComparisonUnitGroup.cs

```diff
@@ -0,0 +1,117 @@
+/// are coalesced can't be correlated to paragraphs in the other document via their hash.  At that point we no longer care what their unids are.
+///
+/// But after that it is only used to reconstruct the tree.  It is also used in the debugging code that
+/// prints the various correlated sequences and comparison units - this is display for debugging purposes only.
+
+/// The key idea here is that a given paragraph will always have the same ancestors, and it doesn't matter whether the content was deleted from the old document,
+/// inserted into the new document, or set as equal.  At this point, we identify a paragraph as a sequential list of content atoms, terminated by a paragraph mark.
+/// This entire list will for a single paragraph, regardless of whether the paragraph is a child of the body, or if the paragraph is in a cell in a table, or if
+/// the paragraph is in a text box.  The list of ancestors, from the paragraph to the root of the XML tree will be the same for all content atoms in the paragraph.
+///
+/// Therefore:
+///
+/// Iterate through the list of content atoms backwards.  When the loop sees a paragraph mark, it gets the ancestor unids from the paragraph mark to the top of the
+/// tree, and sets this as the same for all content atoms in the paragraph.  For descendants of the paragraph mark, it doesn't really matter if content is put into
+/// separate runs or what not.  We don't need to be concerned about what the unids are for descendants of the paragraph.
+
+namespace OpenXmlPowerTools
+{
+    internal class ComparisonUnitGroup : ComparisonUnit
+    {
+        public ComparisonUnitGroupType ComparisonUnitGroupType;
+        public string CorrelatedSHA1Hash;
+        public string StructureSHA1Hash;
+
+        public ComparisonUnitGroup(IEnumerable<ComparisonUnit> comparisonUnitList, ComparisonUnitGroupType groupType, int level)
+        {
+            Contents = comparisonUnitList.ToList();
+            ComparisonUnitGroupType = groupType;
+            var first = comparisonUnitList.First();
+            var comparisonUnitAtom = GetFirstComparisonUnitAtomOfGroup(first);
+            XName ancestorName = null;
+            if (groupType == OpenXmlPowerTools.ComparisonUnitGroupType.Table)
+            {
+                ancestorName = W.tbl;
+            }
+            else if (groupType == OpenXmlPowerTools.ComparisonUnitGroupType.Row)
+            {
+                ancestorName = W.tr;
+            }
+            else if (groupType == OpenXmlPowerTools.ComparisonUnitGroupType.Cell)
+            {
+                ancestorName = W.tc;
+            }
+            else if (groupType == OpenXmlPowerTools.ComparisonUnitGroupType.Paragraph)
+            {
+                ancestorName = W.p;
+            }
+            else if (groupType == OpenXmlPowerTools.ComparisonUnitGroupType.Textbox)
+            {
+                ancestorName = W.txbxContent;
```

Comment:
Codacy found an issue: [Remove this useless assignment to local variable 'ancestorName'.](https://app.codacy.com/gh/Codeuctivity/Open-Xml-PowerTools/pullRequest?prid=6310620)